[Bug]: Dataflow templates test suite running against Beam snapshots is failing

### Related Template(s)

multiple

### Template Version

latest

### What happened?

GHA: https://github.com/GoogleCloudPlatform/DataflowTemplates/actions/workflows/run-it-tests-beam-snapshots.yml

Faliures:

```
14:24:56 ::Downloads$ cat job-logs2.txt | grep "<<< FAILURE"
2026-02-11T18:43:06.0940068Z [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.454 s <<< FAILURE! -- in com.google.cloud.teleport.v2.spanner.service.SpannerServiceTest
2026-02-11T18:43:16.2302904Z [ERROR] Tests run: 8, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 1.356 s <<< FAILURE! -- in com.google.cloud.teleport.v2.spanner.service.SpannerServiceTest
2026-02-11T18:44:23.2252106Z [ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 3.645 s <<< FAILURE! -- in com.google.cloud.teleport.v2.templates.BigQueryToParquetTest
2026-02-11T18:44:33.8697805Z [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.129 s <<< FAILURE! -- in com.google.cloud.teleport.v2.spanner.exceptions.SpannerExceptionParserTest
2026-02-11T18:44:39.4032884Z [ERROR] Tests run: 12, Failures: 0, Errors: 12, Skipped: 0, Time elapsed: 0.730 s <<< FAILURE! -- in com.google.cloud.teleport.v2.spanner.testutils.failureinjectiontesting.NetworkFailureInjectorTest
2026-02-11T18:44:43.0038708Z [ERROR] Tests run: 3, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 0.905 s <<< FAILURE! -- in com.google.cloud.teleport.v2.spanner.testutils.failureinjectiontesting.DataflowFailureInjectorTest
2026-02-11T18:45:11.4288541Z [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 16.23 s <<< FAILURE! -- in com.google.cloud.teleport.v2.transforms.GoogleAdsRowToReportRowJsonFnTest
2026-02-11T18:45:11.5527004Z [ERROR] Tests run: 2, Failures: 1, Errors: 1, Skipped: 0, Time elapsed: 0.088 s <<< FAILURE! -- in com.google.cloud.teleport.v2.utils.GoogleAdsUtilsTest
2026-02-11T18:45:15.0783501Z [ERROR] Tests run: 8, Failures: 0, Errors: 8, Skipped: 0, Time elapsed: 0.049 s <<< FAILURE! -- in com.google.cloud.teleport.v2.utils.FirestoreConvertersTest
2026-02-11T18:45:19.3254639Z [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.637 s <<< FAILURE! -- in com.google.cloud.teleport.v2.templates.DLPTextToBigQueryStreamingTest
2026-02-11T18:45:23.1756935Z [ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.090 s <<< FAILURE! -- in com.google.cloud.teleport.v2.templates.bigtablechangestreamstopubsub.schemautils.PubSubUtilsTest
2026-02-11T18:45:24.4950917Z [ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 1.314 s <<< FAILURE! -- in com.google.cloud.teleport.v2.templates.SpannerChangeStreamsToGcsTest
2026-02-11T18:45:24.6013607Z [ERROR] Tests run: 3, Failures: 0, Errors: 1, Skipped: 2, Time elapsed: 0.096 s <<< FAILURE! -- in com.google.cloud.teleport.v2.templates.TextIOToBigQueryTest
2026-02-11T18:45:46.3815947Z [ERROR] Tests run: 6, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 15.39 s <<< FAILURE! -- in com.google.cloud.teleport.v2.templates.PubsubProtoToBigQueryTest
2026-02-11T18:45:58.5232310Z [ERROR] Tests run: 3, Failures: 0, Errors: 3, Skipped: 0, Time elapsed: 4.498 s <<< FAILURE! -- in com.google.cloud.teleport.v2.templates.JdbcToBigQueryTest
2026-02-11T18:46:02.3543175Z [ERROR] Tests run: 8, Failures: 0, Errors: 8, Skipped: 0, Time elapsed: 0.400 s <<< FAILURE! -- in com.google.cloud.teleport.v2.templates.JdbcToBigQueryReadWithPartitionsTest
2026-02-11T18:46:24.1155976Z [ERROR] Tests run: 10, Failures: 0, Errors: 10, Skipped: 0, Time elapsed: 0.657 s <<< FAILURE! -- in com.google.cloud.teleport.v2.templates.DataplexBigQueryToGcsTest
```

Sample errors:

```
2026-02-11T18:45:27.1291642Z [ERROR] Errors: 
2026-02-11T18:45:27.1294159Z [ERROR]   DLPTextToBigQueryStreamingTest.testFileIOToBigQueryStreamingE2E:144 » ExceptionInInitializer
2026-02-11T18:45:27.1296865Z [ERROR]   SpannerChangeStreamsToGcsTest.testFileFormatFactoryInvalid » ExceptionInInitializer
2026-02-11T18:45:27.1300452Z [ERROR]   SpannerChangeStreamsToGcsTest.testInvalidWindowDuration » NoClassDefFound Could not initialize class com.google.cloud.spanner.spi.v1.SpannerErrorInterceptor
2026-02-11T18:45:27.1303282Z [ERROR]   TextIOToBigQueryTest.setUp:83 » ExceptionInInitializer
2026-02-11T18:45:27.1305406Z [ERROR]   PubSubUtilsTest.testMapChangeJsonStringToPubSubMessageAsJsonValid:79 » ExceptionInInitializer
2026-02-11T18:45:27.1307872Z [ERROR]   FirestoreConvertersTest.testData:65 NoClassDefFound Could not initialize class com.google.datastore.v1.Entity
2026-02-11T18:45:27.1311662Z [ERROR]   FirestoreConvertersTest.testData:65 NoClassDefFound Could not initialize class com.google.datastore.v1.Entity
2026-02-11T18:45:27.1313245Z [ERROR]   FirestoreConvertersTest.testData:65 ExceptionInInitializer
2026-02-11T18:45:27.1314778Z [ERROR]   FirestoreConvertersTest.testData:65 NoClassDefFound Could not initialize class com.google.datastore.v1.Entity
2026-02-11T18:45:27.1317065Z [ERROR]   FirestoreConvertersTest.testData:65 NoClassDefFound Could not initialize class com.google.datastore.v1.Entity
2026-02-11T18:45:27.1320010Z [ERROR]   FirestoreConvertersTest.testData:65 NoClassDefFound Could not initialize class com.google.datastore.v1.Entity
2026-02-11T18:45:27.1322127Z [ERROR]   FirestoreConvertersTest.testData:65 NoClassDefFound Could not initialize class com.google.datastore.v1.Entity
2026-02-11T18:45:27.1324213Z [ERROR]   FirestoreConvertersTest.testData:65 NoClassDefFound Could not initialize class com.google.datastore.v1.Entity
```


```
2026-02-10T00:23:28.3072965Z java.lang.reflect.InaccessibleObjectException: Unable to make static boolean java.nio.Bits.unaligned() accessible: module java.base does not "opens java.nio" to unnamed module @51a9ad5e
2026-02-10T00:23:28.3075245Z 	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
2026-02-10T00:23:28.3076787Z 	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
2026-02-10T00:23:28.3077858Z 	at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:200)
2026-02-10T00:23:28.3078712Z 	at java.base/java.lang.reflect.Method.setAccessible(Method.java:194)
2026-02-10T00:23:28.3079855Z 	at org.apache.hadoop.hbase.unsafe.HBasePlatformDependent.checkUnaligned(HBasePlatformDependent.java:176)
2026-02-10T00:23:28.3081165Z 	at org.apache.hadoop.hbase.unsafe.HBasePlatformDependent.<clinit>(HBasePlatformDependent.java:49)
2026-02-10T00:23:28.3082676Z 	at org.apache.hadoop.hbase.util.Bytes.<clinit>(Bytes.java:130)
2026-02-10T00:23:28.3084165Z 	at com.google.cloud.teleport.bigtable.BigtableToVectorEmbeddingsTest.testBigtableToVectorEmbeddings_restricts(BigtableToVectorEmbeddingsTest.java:202)
2026-02-10T00:23:28.3087138Z 	at com.google.cloud.teleport.bigtable.BigtableToVectorEmbeddingsTest.testBigtableToVectorEmbeddings_restricts(BigtableToVectorEmbeddingsTest.java:144)
2026-02-10T00:23:28.3090307Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2026-02-10T00:23:28.3092774Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2026-02-10T00:23:28.3095292Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2026-02-10T00:23:28.3097511Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
2026-02-10T00:23:28.3099092Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
2026-02-10T00:23:28.3101054Z 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2026-02-10T00:23:28.3103068Z 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
2026-02-10T00:23:28.3105461Z 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2026-02-10T00:23:28.3106459Z 	at org.apache.beam.sdk.testing.TestPipeline$1.evaluate(TestPipeline.java:369)
2026-02-10T00:23:28.3107337Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2026-02-10T00:23:28.3108238Z 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
2026-02-10T00:23:28.3109133Z 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
2026-02-10T00:23:28.3110141Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
2026-02-10T00:23:28.3111166Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
2026-02-10T00:23:28.3112602Z 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
2026-02-10T00:23:28.3113486Z 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
2026-02-10T00:23:28.3114281Z 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
2026-02-10T00:23:28.3115077Z 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
2026-02-10T00:23:28.3115839Z 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
2026-02-10T00:23:28.3116598Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2026-02-10T00:23:28.3117351Z 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
2026-02-10T00:23:28.3118002Z 	at org.junit.runners.Suite.runChild(Suite.java:128)
2026-02-10T00:23:28.3118576Z 	at org.junit.runners.Suite.runChild(Suite.java:27)
2026-02-10T00:23:28.3119233Z 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
2026-02-10T00:23:28.3120024Z 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
2026-02-10T00:23:28.3120797Z 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
2026-02-10T00:23:28.3121593Z 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
2026-02-10T00:23:28.3123035Z 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
2026-02-10T00:23:28.3123821Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2026-02-10T00:23:28.3124670Z 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
2026-02-10T00:23:28.3125448Z 	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:49)
2026-02-10T00:23:28.3126563Z 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:120)
2026-02-10T00:23:28.3127839Z 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:95)
2026-02-10T00:23:28.3129014Z 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
2026-02-10T00:23:28.3130181Z 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:69)
2026-02-10T00:23:28.3131301Z 	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:146)
2026-02-10T00:23:28.3133436Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
2026-02-10T00:23:28.3135671Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
2026-02-10T00:23:28.3137244Z 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
2026-02-10T00:23:28.3138183Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
2026-02-10T00:23:28.4061605Z [ERROR] Tests run: 6, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 13.99 s <<< FAILURE! -- in com.google.cloud.teleport.v2.templates.PubsubProtoToBigQueryTest

```

```
2026-02-11T18:44:52.4808806Z [ERROR] Errors: 
2026-02-11T18:44:52.4810416Z [ERROR]   SpannerExceptionParserTest.<clinit>:32 » ExceptionInInitializer
2026-02-11T18:44:52.4811848Z [ERROR]   DataflowFailureInjectorTest.testAbruptlyKillWorkers_operationHasError_logsError:106 NoClassDefFound Could not initialize class com.google.cloud.compute.v1.Instance
2026-02-11T18:44:52.4813532Z [ERROR]   DataflowFailureInjectorTest.testAbruptlyKillWorkers_stopsCorrectVms:64 ExceptionInInitializer
2026-02-11T18:44:52.4815100Z [ERROR] com.google.cloud.teleport.v2.spanner.testutils.failureinjectiontesting.NetworkFailureInjectorTest.blockNetwork_egressCreationFails
2026-02-11T18:44:52.4816769Z [ERROR]   Run 1: NetworkFailureInjectorTest.setUp:68 NoClassDefFound Could not initialize class com.google.cloud.compute.v1.Operation
2026-02-11T18:44:52.4819966Z [ERROR]   Run 2: NetworkFailureInjectorTest.blockNetwork_egressCreationFails » UnfinishedStubbing 
2026-02-11T18:44:52.4820765Z Unfinished stubbing detected here:
2026-02-11T18:44:52.4821993Z -> at com.google.cloud.teleport.v2.spanner.testutils.failureinjectiontesting.NetworkFailureInjectorTest.setUp(NetworkFailureInjectorTest.java:68)
2026-02-11T18:44:52.4823222Z 
2026-02-11T18:44:52.4823410Z E.g. thenReturn() may be missing.
2026-02-11T18:44:52.4823803Z Examples of correct stubbing:
2026-02-11T18:44:52.4824263Z     when(mock.isOk()).thenReturn(true);
2026-02-11T18:44:52.4825114Z     when(mock.isOk()).thenThrow(exception);
2026-02-11T18:44:52.4825719Z     doThrow(exception).when(mock).someVoidMethod();
```

### Relevant log output

```shell

```

--- a/.github/workflows/run-it-tests-beam-snapshots.yml
+++ b/.github/workflows/run-it-tests-beam-snapshots.yml
@@ -29,7 +29,12 @@ concurrency:
   cancel-in-progress: true
 
 env:
-  MAVEN_OPTS: -Dorg.slf4j.simpleLogger.log.org.apache.maven.plugins.shade=error
+  MAVEN_OPTS: >-
+    -Dorg.slf4j.simpleLogger.log.org.apache.maven.plugins.shade=error
+    --add-opens java.base/java.nio=ALL-UNNAMED
+    --add-opens java.base/sun.nio.ch=ALL-UNNAMED
+    --add-opens java.base/java.lang=ALL-UNNAMED
+    --add-opens java.base/java.util=ALL-UNNAMED
 
 permissions:
   contents: write # Required to modify pom.xml and then revert it

